1 - DataTypes/Code/WeirdCalculator.cs

```diff
@@ -39,29 +41,47 @@ public static decimal ComplexDecimalAdd(float firstValue, double secondValue, de
 			 * I will give you one clue / tip - You *will* need to handle exceptions for the final solution.
 			 * If you are new to exceptions, google "C# Exception Handling" - litterally half our jobs is googling :P
 			 */
-
-			throw new NotImplementedException();
+            decimal firstValueDecimal;
+            decimal secondValueDecimal;
+             try
+            {
+                 firstValueDecimal = (decimal)firstValue;
+                 secondValueDecimal = (decimal)secondValue;
+            }
+                catch (System.OverflowException) 
+            {
+                return 0; 
+            }
+            var result = firstValueDecimal + secondValueDecimal + thirdValue;
+            return result;
+			
 		}
 
 		public static decimal ComplexAdd(char firstValue, string secondValue)
 		{
-			/*
+            /*
 			 * You will notice the TDD tests around this just "assume" it will work. Do not worry about handling exceptions or errors, assume input is valid.
 			 */
-
-			throw new NotImplementedException();
+            decimal x = firstValue - '0';
+            var result = x + Convert.ToDecimal(secondValue);
+            return result;
```

Comment:
For the first value, you can either convert it to a number using the 'Char' object or by using the Convert object and doing firstValue.ToSring()